pianoPlayer/main.js

```diff
@@ -0,0 +1,108 @@
+/* ISSUES & TODOs
+- add music!!
+- stop propagation / capture=true for when the keys are pressed (excluding the letters!)
+- mouseup event not caught when mouseout before
+- prevent key from overlapping the dark keys when it's being pressed
+- fix skeuomorphic design for upper keys
+*/
+
+// The keys and notes variables store the piano keys
+// plucks out the corresponding DOM elements and stores them as such
+const keys = ['c-key', 'd-key', 'e-key', 'f-key', 'g-key', 'a-key', 'b-key', 'high-c-key', 'c-sharp-key', 'd-sharp-key', 'f-sharp-key', 'g-sharp-key', 'a-sharp-key'];
+const notes = [];
+keys.forEach(function (key) {
+    notes.push(document.getElementById(key));
+})
+
+// simple skeuomorphic design for when a key is played
+function keyPlay(event) {
+    event.target.style.backgroundColor = 'darkgrey';
+    event.target.style.position = 'relative';
+    event.target.style.top = '5px';
+    event.target.style.left = '5px';
+}
+function keyReturn(event) {
+    event.target.style.backgroundColor = '';
+    event.target.style.position = '';
+    event.target.style.top = '';
+    event.target.style.left = '';
+}
+function playNote(note) {
+    note.addEventListener('mousedown', keyPlay);
+    note.addEventListener('mouseup', keyReturn);
+}
```

Comment:
Nit: I would recommend different names for these kinds of variables.

Consider the verbs you're using. If I ran a function named `playNote`, I would expect the not to be played.

But that's not what this is doing, it's attaching the event listeners to allow the user to play the note.

So I'd rename these methods to maybe something like:
```ts
function applyPlayKeyStyles

function unapplyPlayKeyStyles

function attachKeyPlayEventListeners
```
Not a big deal overall, but precise/accurate naming of variables/methods like this will make it much easier to reason about your code.

There's a concept called "code as documentation" that I realize is pertinent here.

Notice this code:
```ts
// loop that assigns the eventListeners to each key DOM element
notes.forEach(playNote);
```
vs
```ts
notes.forEach(attachKeyPlayEventListeners);
```

It should be apparent that through better variable names, we've removed the need for the comment entirely. It 'documents' itself.

This is the path towards elegant JS (and code in general).

Reply:
Ok, totally agree! I will remember that!